diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/IndentationRule.kt

```diff
@@ -0,0 +1,206 @@
+import org.cqfn.diktat.ruleset.rules.getDiktatConfigRules
+import org.cqfn.diktat.ruleset.utils.findAllNodesWithSpecificType
+import org.cqfn.diktat.ruleset.utils.getAllLLeafsWithSpecificType
+import org.cqfn.diktat.ruleset.utils.indentBy
+import org.cqfn.diktat.ruleset.utils.indentation.AssignmentOperatorChecker
+import org.cqfn.diktat.ruleset.utils.indentation.CustomIndentationChecker
+import org.cqfn.diktat.ruleset.utils.indentation.DotCallChecker
+import org.cqfn.diktat.ruleset.utils.indentation.ExpressionIndentationChecker
+import org.cqfn.diktat.ruleset.utils.indentation.IndentationConfig
+import org.cqfn.diktat.ruleset.utils.indentation.KDocIndentationChecker
+import org.cqfn.diktat.ruleset.utils.indentation.SuperTypeListChecker
+import org.cqfn.diktat.ruleset.utils.indentation.ValueParameterListChecker
+import org.cqfn.diktat.ruleset.utils.leaveOnlyOneNewLine
+import org.cqfn.diktat.ruleset.utils.log
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.PsiWhiteSpace
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.PsiWhiteSpaceImpl
+import org.jetbrains.kotlin.psi.psiUtil.parents
+import org.jetbrains.kotlin.psi.psiUtil.startOffset
+import org.jetbrains.kotlin.utils.addToStdlib.firstNotNullResult
+
+/**
+ * Rule that checks indentation. The following general rules are checked:
+ * 1. Only spaces should be used each indentation is equal to 4 spaces
+ * 2. File should end with new line
+ * Additionally, a set of CustomIndentationChecker objects checks all WHITE_SPACE node if they are exceptions from general rules.
+ * @see CustomIndentationChecker
+ */
+class IndentationRule : Rule("indentation") {
+    companion object {
+        const val INDENT_SIZE = 4
+        private val increasingTokens = listOf(LPAR, LBRACE, LBRACKET)
+        private val decreasingTokens = listOf(RPAR, RBRACE, RBRACKET)
+    }
+
+    private lateinit var configuration: IndentationConfig
+    private lateinit var customIndentationCheckers: List<CustomIndentationChecker>
+
+    private lateinit var configRules: List<RulesConfig>
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+    private var fileName: String = ""
+
+    override fun visit(node: ASTNode, autoCorrect: Boolean, params: KtLint.Params, emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        configRules = params.getDiktatConfigRules()
+        isFixMode = autoCorrect
+        emitWarn = emit
+        fileName = params.fileName!!
+
+        configuration = IndentationConfig(configRules.getRuleConfig(WRONG_INDENTATION)?.configuration
+                ?: mapOf())
+        customIndentationCheckers = listOf(
+                AssignmentOperatorChecker(configuration),
```

Comment:
do you check lambdas?
a newline after ->

Reply:
Not explicitly; every opening brace causes increase of indentation, and as a consequence body of lambda is indented.